install/outline-install.sh

```diff
@@ -0,0 +1,157 @@
+
+read -r -p "Enter the public url for your Outline instance (e.g., https://outline.your-domain.tld)): " URL
+read -r -p "Would you like to add Adminer? <y/N> " adminer_prompt
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  sudo \
+  curl \
+  lsb-release \
+  postgresql \
+  gnupg \
+  unzip
+msg_ok "Installed Dependencies"
+
+msg_info "Setting up Node.js Repository"
+NODE_MAJOR=20
+mkdir -p /etc/apt/keyrings
+wget -qO- https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_MAJOR}.x nodistro main" >/etc/apt/sources.list.d/nodesource.list
+msg_ok "Set up Node.js Repository"
+
+msg_info "Setting up Redis Repository"
+wget -qO- https://packages.redis.io/gpg | gpg --dearmor -o /etc/apt/keyrings/redis-archive-keyring.gpg
+echo "deb [signed-by=/etc/apt/keyrings/redis-archive-keyring.gpg] https://packages.redis.io/deb $(lsb_release -cs) main" >/etc/apt/sources.list.d/redis.list
+msg_ok "Set up Redis Repository"
+
+msg_info "Installing Node.js/Yarn"
+$STD apt-get update
+$STD apt-get install -y nodejs
+$STD npm install -g yarn
+msg_ok "Installed Node.js/Yarn"
+
+msg_info "Installing Redis"
+$STD apt-get install -y redis
+sed -i 's/^bind .*/bind 0.0.0.0/' /etc/redis/redis.conf
+systemctl enable -q --now redis-server.service
+msg_ok "Installed Redis"
+
+msg_info "Setting up PostgreSQL DB"
+SECRET_KEY="$(openssl rand -hex 32)"
+UTILS_SECRET="$(openssl rand -hex 32)"
+DB_NAME=outlinedb
+DB_USER=outline
+DB_PASS="$(openssl rand -base64 18 | tr -d '/' | cut -c1-13)"
+DATABASE_URL="postgres://${DB_USER}:${DB_PASS}@localhost:5432/${DB_NAME}"
+$STD sudo -u postgres psql -c "CREATE ROLE $DB_USER WITH LOGIN PASSWORD '$DB_PASS';"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER WITH CREATEDB;"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER SET timezone TO 'UTC';"
+
+{
+    echo "Outline-Credentials"
+    echo "Outline Database User: $DB_USER"
+    echo "Outline Database Password: $DB_PASS"
+    echo "Outline Database Name: $DB_NAME"
+    echo "Outline Secret: $SECRET_KEY"
+    echo "Outline Utils Secret: $UTILS_SECRET"
+} >> ~/outline.creds
+msg_ok "Set up PostgreSQL DB"
+
+
+if [[ "${adminer_prompt,,}" =~ ^(y|yes)$ ]]; then
+  msg_info "Installing Adminer"
+  $STD apt install -y adminer
+  $STD a2enconf adminer
+  systemctl reload apache2
+  IP=$(hostname -I | awk '{print $1}')
+  {
+      echo ""
+      echo "Adminer Interface: $IP/adminer/"
+      echo "Adminer System: PostgreSQL"
+      echo "Adminer Server: localhost:5432"
+      echo "Adminer Username: $DB_USER"
+      echo "Adminer Password: $DB_PASS"
+      echo "Adminer Database: $DB_NAME"
+  } >> ~/outline.creds
+  msg_ok "Installed Adminer"
+fi
```

Comment:
Does this area work? Haven't seen this installation command being used anywhere off the top of my head?

Reply:
I modfied this from https://github.com/community-scripts/ProxmoxVE/blob/fab35a61902384a1425f6211a6511f489cc42d69/install/linkwarden-install.sh#L64-L88, and tested that it works.